crates/bevy_ecs/src/storage/resource.rs

```diff
@@ -30,18 +31,32 @@ impl ResourceData {
     /// Gets a read-only pointer to the underlying resource, if available.
     #[inline]
     pub fn get_data(&self) -> Option<Ptr<'_>> {
-        self.column.get_data(Self::ROW)
+        self.data.get_ptr()
     }
 
     /// Gets a read-only reference to the change ticks of the underlying resource, if available.
     #[inline]
     pub fn get_ticks(&self) -> Option<ComponentTicks> {
-        self.column.get_ticks(Self::ROW)
+        // SAFETY: If the data is present, the ticks have been written to with valid values
```

Comment:
This comment isn't right, it needs to consider the safety invariants of `UnsafeCellDeref::read`.